ATSQuadroStrategyBase/NinjaTrader 8/bin/Custom/Strategies/AlgoSystemBase.cs

```diff
@@ -2029,9 +2029,9 @@ public void SubmitSignalAction(AlgoSignalAction AlgoSignalActions, string signal
         {
             if (tracing) Print("SubmitSignalAction " + AlgoSignalActions.ToString() + " WF=" + this.TradeWorkFlow.ToString());
 
-            TEQ.Enqueue(new AlgoSignalActionMsq(AlgoSignalActions, IsHistoricalTradeOrPlayBack ? Time[0] : Now, signalLabel));

```

Comment:
TEQ.Enqueue(new AlgoSignalActionMsq(AlgoSignalActions, IsHistoricalTradeOrPlayBack ? Time[0] : Now, signalLabel));
this is required for historical we are using the bar time - in realtime we are using realtime

Reply:
With the Now fix, however shouldn't Time[0] and Now be relatively the same, just one is the bar close time, e.g. delta depends on what `Calculate` is set to, Close, Price, Tick, etc...  either way
```
IsHistorical ? Time[0] : Now
```
would perhaps be the better option.

With the fixes to Now() Playback mode doesn't have issues with timeouts etc, because it's sense of time is in sync with data time.   Which I think solves a lot of the issues.

The fact that things like OrdersActive List is not populated in "Unsafe Mode" is a bug and other differing behaviors makes me prefer PlayBack uses the full code path...  I feel like the Now bug was causing many issues in Historical/Playback that may have impacted some of these decisions in the past.

I'm not seeing a significant performance difference with the additional checks, but I haven't done any actual time comparisons.

Reply:
":With the Now fix, however shouldn't Time[0] and Now be relatively the same"
no not at all - Time[0] is bar Time - now is an arbitary date which can be now in reality or now in a playback  universe and not related to bar close Time[0]   - i prefer to have 3 dates for usage for various reasons which bvecome apparent in different layers later:
1) bar time - replay and backtest use this as does realtime where appropiate
2) time replay  versus time local time  

"The fact that things like OrdersActive List is not populated in "Unsafe Mode" is a bug"
Unsafe mode skips checks that is its purpose  so we would not need to slow down to use certain flows and objects. by design.  Unsafe mode works in testing without error so we have seen no bugs related to that - unless we missed one etc.
But surely if there was- so we can raise this as a bug and then investigate.

**With the fixes to Now() Playback mode doesn't have issues with timeouts etc,**
What issues none were reported no bugs reported yet with playback or unsafe.
This change was approved as it should be inocous on the workflow and so on 

"I'm not seeing a significant performance difference with the additional checks, but I haven't done any actual time comparisons." 
Playback is for 1 to 1000 Times speed -   Playback does not need the additional checks this is by design -or you start to use parts of the system which are reserved for realtime only.   

**because it's sense of time is in sync with data time. Which I think solves a lot of the issues.**
There were no issues reported in the unchanged codebase for playback.
- if you change the structure of the code to use the workflow certainly timeouts will occur and caveats.
What was  the reason for changing the code and getting the caveats issues ?

Reply:
This is a probably a better place to ask so i can get on same page and understand why such broad changes are proposed wtih unknown impact and large regressions testing rand impact analysis and goes against the design principle to keep it lean and mean:    #95